state/model.go

```diff
@@ -281,7 +279,7 @@ func (m ModelArgs) Validate() error {
 	if m.Type == modelTypeNone {
 		return errors.NotValidf("empty Type")
 	}
-	if m.Type == ModelTypeCAAS && !featureflag.Enabled(feature.CAAS) {
+	if m.Type == ModelTypeCAAS {
```

Comment:
I think the whole conditional should go away.
This would make CAAS types invalid always...